nabak/toro-fullstack
# Feature/postgres persistence

[Feature] Add Docker support and database setup

This pull request adds support for Docker containers for both the API and the database, as well as creates the necessary database schema and inserts initial data. The mock data has also been removed and replaced with real data.
I have also created the necessary services and persistence for the userPosition endpoint. This endpoint allows the user to view their current balance, list of assets they own, and their consolidated net worth. The getUserPosition method in the UserPositionService class retrieves the user's account information from the AccountPersistence class and calculates their net worth by summing up the balance in their checking account and the current value of their assets. The endpoint returns a UserPositionResponse object containing this information.

## Changes Made
- Added Docker support for the API
- Created a Docker container for the database
- Set up the database schema
- Inserted initial data into the database
- Removed mock data and replaced with real data
- Created the necessary services and persistence for the user position endpoint.

## How to Test
1. Clone the repository and navigate to the project directory
2. Install Docker and Docker Compose
3. Run `docker-compose up` to start the containers
4. Send requests to the API to verify that it is working as expected
